vip_hci/hci_dataset.py

```diff
@@ -789,16 +793,101 @@ def inject_companions(self, flux, rad_dists, n_branches=1, theta=0,
-                                               rad_dists, n_branches, theta,
-                                               imlib, interpolation,
-                                               full_output=True,
-                                               verbose=verbose)
+        self.cube, yx = cube_inject_companions(
+            self.cube, self.psfn, self.angles, flux, self.px_scale,
+            rad_dists, n_branches, theta, imlib, interpolation,
+            full_output=True, verbose=verbose
+        )
+
+        if not hasattr(self, "injections_yx"):
+            self.injections_yx = []
+
+        self.injections_yx += yx
+
+        if verbose:
+            print("Coordinates of the injections stored in self.injections_yx")
 
         if full_output:
             return yx
 
+    def copies_with_injections(self, n_copies, inrad=8, outrad=12,
+                               dist_flux=("uniform", 2, 500)):
+        """
+        Create copies of this dataset, containing different random injections.
+
+        Parameters
+        ----------
+        n_copies : int
+            This is the number of 'cube copies' returned.
+        inrad,outrad : float
+            Inner and outer radius of the injections. The actual injection
+            position is chosen randomly.
+        dist_flux : tuple('method', *params)
+            Tuple describing the flux selection. Method can be a function, the
+            ``*params`` are passed to it. Method can also be a string, for a
+            pre-defined random function:
+
+                ``("skewnormal", skew, mean, var)``
+                    uses scipy.stats.skewnorm.rvs
+                ``("uniform", low, high)``
+                    uses np.random.uniform
+                ``("normal", loc, scale)``
+                    uses np.random.normal
+
+        Returns
+        -------
+        fake_datasets : list of HCIDataset
+            List containing ``n_copies`` fake datasets.
+
+        """
+
+        fake_datasets = []
+
+        res = cube_copies_with_injections(self.cube, self.psf, self.angles,
+                                          self.px_scale, n_copies=n_copies,
+                                          inrad=inrad, outrad=outrad,
+                                          dist_flux=dist_flux)
+
+        for r in res:
+            dsi = self.copy()
+            dsi.cube = r["cube"]
+            dsi.injections_yx = r["position"]
+            datasets.append(dsi)
+            # dist, theta, and flux are not stored.
+
+        return fake_datasets
+
+
+
+
+    def copy(self, deep=True):
+        """
+        Create an in-memory copy of this HCIDataset.
+
+        This is especially useful when creating a 
+
+        Parameters
+        ----------
+        deep : bool, optional
+            By default, a deep copy is created. That means every (sub)attribute
+            is copied in memory. While this requires more memory, one can safely
```

Comment:
Again, we should think about memory management

Reply:
👍